internal/controller/genericprovider_controller_test.go

```diff
@@ -50,7 +50,7 @@ apiVersion: apps/v1
 kind: Deployment
 metadata:
   labels:
-    cluster.x-k8s.io/provider: infrastructure-docker
+    cluster.x-k8s.io/provider: cluster-api
```

Comment:
why do we need this change, can you please elaborate? since deployment is for capd-controller-manager, it should perhaps stay as is

Reply:
Yes, good catch.
We need to change a whole test Deployment manifest to the correct Cluster API Core one.
Before this fix, it was able to deploy any Cluster API manifest with any type of the Operator entity, afterwards updates will be never applied, as clusterctl will be validating this case.
After this fix, applying objects is validating before install.